mkinitcpio

```diff
@@ -211,25 +211,33 @@ build_image() {
             ;;
     esac
 
-    cpio_opts=('-0' '-o' '-H' 'newc')
-    (( _optquiet )) && cpio_opts+=('--quiet')
+    cpio_opts=('--null' '-cf-' '--format=newc')
+    (( _optquiet )) || cpio_opts+=('-v')
     if (( EUID != 0 )); then
         warning 'Not building as root, ownership cannot be preserved'
-        cpio_opts+=('-R' '0:0')
+        cpio_opts+=('--uid' '0' '--gid' '0')
     fi
 
     pushd "$BUILDROOT" >/dev/null
-    find . -mindepth 1 -printf '%P\0' |
-            LANG=C bsdcpio "${cpio_opts[@]}" |
+
+    # Reproducibility: set timestamps if SOURCE_DATE_EPOCH present
+    find . -mindepth 1 -printf '%P\0' | xargs -0 touch -hcd "@${SOURCE_DATE_EPOCH:-$(date +%s)}"
```

Comment:
No need for xargs here, just use `find ... -exec touch -hcd "$SOURCE_DATE_EPOCH" {} +`

Reply:
ack